js/src/UXCapture.js

```diff
@@ -0,0 +1,78 @@
+import ExpectedMark from "./ExpectedMark";
+import View from "./View";
+
+export default class UXCapture {
+  static attachTo(window) {
+    // allow running in Node.js environment
+    if (typeof window === "undefined") {
+      throw new Error("Must provide a valid window object");
+    }
+
+    // already have UX Capture object defined, reuse it
+    if (typeof window.UX !== "undefined") {
+      return window.UX;
+    }
+
+    // assign singleton instance to window object
+    window.UX = new UXCapture(window);
+
+    return window.UX;
+  }
+
+  constructor() {
+    this.onMark = null;
+    this.onMeasure = null;
+  }
+
+  /**
+   * Sets expected marks and corresponding zones for current view
+   *
+   * @TODO re-evaluate if we should allow multiple executions of this method
+   */
+  expect(zoneConfigs) {
+    if (typeof zoneConfigs === "undefined" || !Array.isArray(zoneConfigs)) {
+      return false;
```

Comment:
sort of trivial, but the `false` return value here is a little unexpected - `expect()` will return `false` or `undefined`?  I think I would recommend printing something with `console.error()` and then `return;` with no value just to keep the return type consistent.

Reply:
Will be throwing an error here, similar to how we do that for a window - there is no point to silently accept `UX.expect()` call without an array passed in.

It's OK to pass an empty array though.